### DISCUSSION TITLE
[SOLVED] how to open ruby files in macos? (for example)

### DISCUSSION THREAD

**Discussion by danielb2** [kovidgoyal/kitty#6314] [General]
- macos

if I use file association for macos for python, it will open it in the editor, but if I do the same in ruby, or shell, it will run it or give me an error

```
Open URL Error

Unknown URL type, cannot open:
  file:///Users/daniel/tmp/foo.rb
Press Enter or Esc to exit
```

I tried following the instructions here https://sw.kovidgoyal.net/kitty/open_actions/ and made a `launch-actions.conf` file with the following contents:

```
protocol file
ext rb,conf,plist,sh
action launch --type=os-window $EDITOR $FILE_PATH
```

however, I'm still getting the same error. What am I missing?

**Comment by kovidgoyal** [comment]
You want launch-actions.conf
https://sw.kovidgoyal.net/kitty/open_actions/#scripting-the-opening-of-files-with-kitty-on-macos

On Mon, May 29, 2023 at 11:18:05AM -0700, Daniel Bretoi wrote:
> - macos
> 
> 
> if I use file association for macos for python, it will open it in the editor, but if I do the same in ruby, or shell, it will run it or give me an error
> 
> ```
> Open URL Error
> 
> Unknown URL type, cannot open:
>   file:///Users/daniel/tmp/foo.rb
> Press Enter or Esc to exit
> ```
> 
> I tried following the instructions here https://sw.kovidgoyal.net/kitty/open_actions/ and made a launch-actions.conf file with the following contents:
> 
> ```
> protocol file
> ext rb,conf,plist,sh
> action launch --type=os-window $EDITOR $FILE_PATH
> ```
> 
> however, I'm still getting the same error. What am I missing?
> 
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6314
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by danielb2** [reply]
that's what I said I'm using above. What's wrong with what I did?

**Comment by kovidgoyal** [comment]
No idea, I can tell you that with current kitty running

kitty +open test.sh

executes test.sh for me on macOS, as expected.

sh files are supported by default.

**Reply by danielb2** [reply]
I want to edit it, not execute.  THat's why I tried this:

```
protocol file
ext rb,conf,plist,sh
action launch --type=os-window $EDITOR $FILE_PATH
```

**Comment by kovidgoyal** [comment]
On Mon, May 29, 2023 at 09:35:36PM -0700, Daniel Bretoi wrote:
> I want to edit it, not execute.  THat's why I tried this:
> 
> ```
> protocol file
> ext rb,conf,plist,sh
> action launch --type=os-window $EDITOR $FILE_PATH
> ```

Adding that to ~/.config/kitty/launch-actions.conf causes test.sh to
open in the editor, as expected.

**Reply by danielb2** [reply]
It edits with `kitty +open test.sh` however in Finder opening with contextual manager, and using `open -a Kitty test.sh` it still executes instead of editing. 
<img width="643" alt="Screen Shot 2023-05-30 at 21 00 15" src="https://github.com/kovidgoyal/kitty/assets/26678/0dc6aa0f-77d2-4e93-80bd-e7e65f5340ed">
<img width="210" alt="Screen Shot 2023-05-30 at 21 00 20" src="https://github.com/kovidgoyal/kitty/assets/26678/2acc4086-2c4c-4191-976e-5102cf29ef1d">

**Comment by kovidgoyal** [comment]
On Tue, May 30, 2023 at 09:00:50PM -0700, Daniel Bretoi wrote:
> It edits with `kitty +open test.sh` however in Finder opening with contextual manager, and using `open -a Kitty test.sh` it still executes instead of editing. 
> <img width="643" alt="Screen Shot 2023-05-30 at 21 00 15" src="https://github.com/kovidgoyal/kitty/assets/26678/0dc6aa0f-77d2-4e93-80bd-e7e65f5340ed">
> <img width="210" alt="Screen Shot 2023-05-30 at 21 00 20" src="https://github.com/kovidgoyal/kitty/assets/26678/2acc4086-2c4c-4191-976e-5102cf29ef1d">

REstart kitty. launch-actions.conf is not live loaded.

**Reply by danielb2** [reply]
that was it. thank you 😅

thanks for everything. calibre + kitty and answering too
